### PR TITLE
[inference] add silu linear fusion for smoothquant llama mlp 

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- add silu linear fusion for smoothquant llama mlp 

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/kernel/cuda_native/csrc/smoothquant/linear.cu [hunk 1]
```diff
@@ -0,0 +1,160 @@
+#include "linear.h"
+#include <cutlass/core_io.h>
```
Comment (yuanheng-zhao) [inline colossalai/kernel/cuda_native/csrc/smoothquant/linear.cu:2]: Wonder if we should add 
```python
// modified from XXX
```

File: tests/test_smoothquant/test_smoothquant_linear.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    HAS_SMOOTHQUANT_CUDA = True
+except ImportError:
+    warnings.warn("CUDA gptq is not installed")
+    HAS_SMOOTHQUANT_CUDA = False
+
+TRITON_CUDA_SUPPORT = version.parse(torch.version.cuda) > version.parse("11.4")
+
+
+@pytest.mark.skipif(
+    not TRITON_CUDA_SUPPORT or not HAS_SMOOTHQUANT_CUDA,
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_smoothquant_linear.py:20]: It does not check whether triton is properly installed

File: tests/test_smoothquant/test_smoothquant_linear.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    reason="triton requires cuda version to be higher than 11.4",
+)
+def test_linear():
+    a = torch.randint(-127, 127, (128, 512), dtype=torch.int8, device="cuda")
+    b = torch.randint(-127, 127, (512, 256), dtype=torch.int8, device="cuda")
+    c = torch.rand(256, dtype=torch.float, device="cuda")
+
+    alpha = 1 / 127
+    beta = 1.0
+    torch_out = torch.mm(a.to(torch.float) * alpha, b.to(torch.float)) + c
```
Comment (yuanheng-zhao) [inline tests/test_smoothquant/test_smoothquant_linear.py:30]: Unused var `torch_out`

File: colossalai/kernel/cuda_native/csrc/smoothquant/linear.cu (hunk 1 above)
Comment (Xu-Kai) [reply colossalai/kernel/cuda_native/csrc/smoothquant/linear.cu:4]: added.

File: tests/test_smoothquant/test_smoothquant_linear.py (hunk 2 above)
Comment (Xu-Kai) [reply tests/test_smoothquant/test_smoothquant_linear.py]: It dose not depend on triton. Update the skip condition to only check CUDA.

Comment (Xu-Kai) [reply tests/test_smoothquant/test_smoothquant_linear.py]: Updated.

**Pull request merged**